ote/src/clj/ote/services/transport.clj

```diff
@@ -123,11 +123,19 @@
 
 (defn- fix-price-classes
   "Frontend sends price classes prices as floating points. Convert them to bigdecimals before db insert."
+  [service data-path]
+  (update-in service [data-path ::t-service/price-classes]
+             (fn [price-classes-float]
+               (mapv #(update % ::t-service/price-per-unit bigdec) price-classes-float))))
+
+(defn- floats-to-bigdec
+  "Frontend sends some values as floating points. Convert them to bigdecimals before db insert."
   [service]
-  (if (= :passenger-transportation (::t-service/type service))
-    (update-in service [::t-service/passenger-transportation ::t-service/price-classes]
-               (fn [price-classes-float]
-                 (mapv #(update % ::t-service/price-per-unit bigdec) price-classes-float)))
+  (case (::t-service/type service)
```

Comment:
First implentation is poor. I'm not sure should we change this now but, that case is not needed. Type can be converted to namespace + keyword.

Reply:
Can you elaborate what you mean?